pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -103,7 +105,15 @@ func (l *LocalPipelineRefResolver) GetPipeline(ctx context.Context, name string)
 	if l.Namespace == "" {
 		return nil, fmt.Errorf("Must specify namespace to resolve reference to pipeline %s", name)
 	}
-	return l.Tektonclient.TektonV1beta1().Pipelines(l.Namespace).Get(ctx, name, metav1.GetOptions{})
+
+	pipeline, err := l.Tektonclient.TektonV1beta1().Pipelines(l.Namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	if err := verifyResolvedPipeline(ctx, pipeline.DeepCopy()); err != nil {
```

Comment:
We are passing a copy of the pipelien but within verifyResolvedPipeline, we are making another copy of the pipeline

Reply:
fixed! Thanks!